editor/src/java/com/defold/editor/DiscordRichPresence.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020-2025 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package com.defold.editor;
+
+import com.google.gson.JsonObject;
+import com.jagrosh.discordipc.IPCClient;
+import com.jagrosh.discordipc.IPCListener;
+import com.jagrosh.discordipc.entities.ActivityType;
+import com.jagrosh.discordipc.entities.Packet;
+import com.jagrosh.discordipc.entities.RichPresence;
+import com.jagrosh.discordipc.entities.User;
+import com.jagrosh.discordipc.exceptions.NoDiscordClientException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.OffsetDateTime;
+
+final class DiscordRichPresence {
+
+    private static final Logger logger = LoggerFactory.getLogger(DiscordRichPresence.class);
+    private static final long CLIENT_ID = 1450057060514402396L;
+    public static final String STATE = "Making a game";
+
+    static void startAsync() {
+        Thread thread = new Thread(DiscordRichPresence::connect, "discord-rich-presence");
+        thread.setDaemon(true);
+        thread.start();
```

Comment:
I think we should use a virtual thread (which is a daemon by default) instead of a daemon platform thread to avoid using too many OS threads.

Maybe something like
```java
Thread.ofVirtual().name("discord-rich-presence").start(DiscordRichPresence::connect);
```